.circleci/osx_install_dependencies.sh

```diff
@@ -43,13 +43,13 @@ then
   ./scripts/install_obsolete_jsoncpp_1_7_4.sh
 
   # z3
```

Comment:
Probably all of this could be replaced by just ``brew install z3`` now. However, then we'd need a different criterion for the outer ``if`` and maybe better to do all this in a separate PR afterwards.

Reply:
Yea, agree